File: scripts/make_dmg_background.py
#!/usr/bin/env python3
"""Generate a DMG background image for mywisper installer."""
import sys
import os

bg_dir = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), "build", "dmg_background")
os.makedirs(bg_dir, exist_ok=True)
bg_path = os.path.join(bg_dir, "background.png")

from AppKit import (
    NSImage, NSMakeSize, NSMakeRect, NSMakePoint,
    NSGradient, NSColor, NSBezierPath, NSFont,
    NSAttributedString, NSFontAttributeName, NSForegroundColorAttributeName,
    NSBitmapImageRep, NSBitmapImageFileTypePNG,
    NSLineCapStyleRound, NSLineJoinStyleRound,
)

width, height = 660, 400

image = NSImage.alloc().initWithSize_(NSMakeSize(width, height))
image.lockFocus()

# Dark gradient background
gradient = NSGradient.alloc().initWithStartingColor_endingColor_(
    NSColor.colorWithRed_green_blue_alpha_(0.08, 0.08, 0.12, 1.0),
    NSColor.colorWithRed_green_blue_alpha_(0.15, 0.15, 0.22, 1.0),
)
gradient.drawInRect_angle_(NSMakeRect(0, 0, width, height), 270)

# Arrow between app icon and Applications
arrowColor = NSColor.colorWithRed_green_blue_alpha_(0.45, 0.45, 0.55, 0.7)
arrowColor.setStroke()

path = NSBezierPath.bezierPath()
path.setLineWidth_(2.5)
path.setLineCapStyle_(NSLineCapStyleRound)
path.moveToPoint_(NSMakePoint(270, height / 2))
path.lineToPoint_(NSMakePoint(390, height / 2))
path.stroke()

# Arrowhead
head = NSBezierPath.bezierPath()
head.setLineWidth_(2.5)
head.setLineCapStyle_(NSLineCapStyleRound)
head.setLineJoinStyle_(NSLineJoinStyleRound)
head.moveToPoint_(NSMakePoint(378, height / 2 + 14))
head.lineToPoint_(NSMakePoint(395, height / 2))
head.lineToPoint_(NSMakePoint(378, height / 2 - 14))
head.stroke()

# Title
titleAttrs = {
    NSFontAttributeName: NSFont.boldSystemFontOfSize_(18),
    NSForegroundColorAttributeName: NSColor.colorWithRed_green_blue_alpha_(0.85, 0.85, 0.92, 1.0),
}
title = NSAttributedString.alloc().initWithString_attributes_("mywisper", titleAttrs)
titleSize = title.size()
title.drawAtPoint_(NSMakePoint((width - titleSize.width) / 2, height - 42))

# Hint text
hintAttrs = {
    NSFontAttributeName: NSFont.systemFontOfSize_(12),
    NSForegroundColorAttributeName: NSColor.colorWithRed_green_blue_alpha_(0.55, 0.55, 0.65, 0.9),
}
hint = NSAttributedString.alloc().initWithString_attributes_("Drag to Applications to install", hintAttrs)
hintSize = hint.size()
hint.drawAtPoint_(NSMakePoint((width - hintSize.width) / 2, 22))

image.unlockFocus()

# Save as PNG
tiffData = image.TIFFRepresentation()
bitmapRep = NSBitmapImageRep.imageRepWithData_(tiffData)
pngData = bitmapRep.representationUsingType_properties_(NSBitmapImageFileTypePNG, {})
pngData.writeToFile_atomically_(bg_path, True)
print(f"Background saved to {bg_path}")

File: scripts/make_dmg_background.swift
#!/usr/bin/env swift
import AppKit

// Oversized to ensure full coverage of Finder window content area
let width: CGFloat = 660
let height: CGFloat = 480
let outputPath = CommandLine.arguments.count > 1
    ? CommandLine.arguments[1]
    : NSString(string: "~").expandingTildeInPath + "/Downloads/dmg_background.png"

let image = NSImage(size: NSSize(width: width, height: height))
image.lockFocus()

// Light solid background (no gradient — clean look)
NSColor(red: 0.93, green: 0.93, blue: 0.95, alpha: 1.0).setFill()
NSRect(x: 0, y: 0, width: width, height: height).fill()

// Arrow between icon positions (app at x=130, Applications at x=390)
// Finder icon y=170 from top → from bottom = height - 170 = 310
let arrowY: CGFloat = height - 170
let arrowColor = NSColor(red: 0.55, green: 0.55, blue: 0.60, alpha: 0.45)
arrowColor.setStroke()

let shaft = NSBezierPath()
shaft.lineWidth = 1.5
shaft.lineCapStyle = .round
shaft.move(to: NSPoint(x: 205, y: arrowY))
shaft.line(to: NSPoint(x: 315, y: arrowY))
shaft.stroke()

// Arrowhead
let head = NSBezierPath()
head.lineWidth = 1.5
head.lineCapStyle = .round
head.lineJoinStyle = .round
head.move(to: NSPoint(x: 305, y: arrowY + 10))
head.line(to: NSPoint(x: 320, y: arrowY))
head.line(to: NSPoint(x: 305, y: arrowY - 10))
head.stroke()

// Hint text at bottom
let hintAttrs: [NSAttributedString.Key: Any] = [
    .font: NSFont.systemFont(ofSize: 11),
    .foregroundColor: NSColor(red: 0.45, green: 0.45, blue: 0.50, alpha: 0.7),
]
let hint = NSAttributedString(string: "Drag to Applications to install", attributes: hintAttrs)
let hintSize = hint.size()
hint.draw(at: NSPoint(x: (width / 2 - hintSize.width / 2) - 20, y: height - 310))

image.unlockFocus()

// Save as PNG
guard let tiffData = image.tiffRepresentation,
      let bitmapRep = NSBitmapImageRep(data: tiffData),
      let pngData = bitmapRep.representation(using: .png, properties: [:]) else {
    print("Failed to create PNG")
    exit(1)
}

let url = URL(fileURLWithPath: outputPath)
try! pngData.write(to: url)
print("Background saved to \(outputPath)")
